runhouse/resources/hardware/on_demand_cluster.py

```diff
@@ -617,6 +617,14 @@ def up(self, verbose: bool = True, force: bool = False):
             logger.info("Provisioning cluster")
             LocalLauncher.up(cluster=self, verbose=verbose)
 
+        if start_server:
+            logger.info("Starting Runhouse server on cluster")
+            self.restart_server()
+
+        if rns_client.autosave_resources():
```

Comment:
I see this was moved from the Launcher? I think we'd always want to save after this when it's happening via Den Launcher so maybe it's better to keep there?

Curious about the reasoning behind the change.